Materials2050_Adapter/Convert/ToBHoM.cs

```diff
@@ -0,0 +1,271 @@
+                            BH.Engine.Base.Compute.RecordError($"The {apiT} API has not yet been implemented, no objects were created from the API query. Please use the OpenAPI.");
+                            break;
+                        }
+                    default:
+                        break;
+                }
+                return epd;
+            }
+            else
+            {
+                BH.Engine.Base.Compute.RecordError("Error.");
+                return null;
+            }
+        }
+
+        /***************************************************/
+
+        public static EnvironmentalProductDeclaration fromOpenAPI(CustomObject co)
+        {
+            EnvironmentalProductDeclaration epd = new EnvironmentalProductDeclaration();
+            // OpenAPI call object schema
+
+            // basic results parsing data
+            ResultsNavigation resultsNavigation = new ResultsNavigation();
+
+            resultsNavigation.TotalProducts = (int)co.PropertyValue("TotalProducts");
+            resultsNavigation.ProductsOnPage = (int)co.PropertyValue("countProductsOnPage");
+            resultsNavigation.CurrentPage = (int)co.PropertyValue("current_page");
+            resultsNavigation.ProductRange = co.PropertyValue("product_range").ToString();
+
+            // Not implemented - no use case provided - inaccessible URLs
+            //string next = co.PropertyValue("next")?.ToString();
+            //string previous = co.PropertyValue("previous")?.ToString();
+
+            // Add additional data where possible 
+            // TODO - The LCA oM fragment is not very well structured and would require significant refactor to be helpful here
+            AdditionalEPDData addData = new AdditionalEPDData();
+            addData.Manufacturer = co.CustomData["company"] as string;
+            addData.Description = co.CustomData["product_type"] as string;
+            addData.Description += $" - {co.CustomData["material_type"] as string}";
+            addData.Manufacturer += $" - {co.CustomData["manufacturing_location"] as string}";
+
+            epd.Name = co.CustomData["name"] as string;
+
+            // Material Facts object within the Results Object
+            CustomObject materialObj = co.CustomData["material_facts"] as CustomObject;
+
+            double unused = double.NaN;
+            double a3 = double.NaN;
+
+            try
+            {
+                a3 = (double)materialObj.CustomData["manufacturing"];
+            }
+            catch { }
+
+            ClimateChangeTotalMetric metric = new ClimateChangeTotalMetric(unused, unused, a3, unused, unused, unused, unused, unused, unused, unused, unused, unused, unused, unused, unused, unused, unused, unused, unused, unused);
+
+            epd.QuantityType = GetQuantityTypeFromString(materialObj.CustomData["declared_unit"] as string);
+            epd.EnvironmentalMetrics.Add(metric);
+
+            // TODO Adding fragments is not working
```

Comment:
What issue are you getting with adding fragments?

Reply:
Ah I fixed this. I was not actually adding the fragments to the object like I was thinking I was doing. I actually had to implement this method like this: 

`EnvironmentalProductDeclaration epdData = (EnvironmentalProductDeclaration)Modify.AddFragment(epd, addData);`

Where I'm making a new epd obj and adding a the fragment (addData) to the existing epd and calling it epdData. Then the same for adding a new fragment like this: 

`EnvironmentalProductDeclaration epdDataNav = (EnvironmentalProductDeclaration)Modify.AddFragment(epdData, resultsNavigation);`

Then I return the modified object with the two fragments attached. 
`return epdDataNav;`